File: src/test/java/io/github/binout/soccer/infrastructure/persistence/InMemoryFriendlyMatchDateRepository.java
package io.github.binout.soccer.infrastructure.persistence;

import io.github.binout.soccer.domain.date.FriendlyMatchDate;
import io.github.binout.soccer.domain.date.FriendlyMatchDateRepository;

import java.time.LocalDate;
import java.time.Month;
import java.util.Comparator;
import java.util.Map;
import java.util.Optional;
import java.util.concurrent.ConcurrentHashMap;
import java.util.stream.Stream;

public class InMemoryFriendlyMatchDateRepository implements FriendlyMatchDateRepository {

    private Map<LocalDate, FriendlyMatchDate> dates;

    public InMemoryFriendlyMatchDateRepository() {
        dates = new ConcurrentHashMap<>();
    }

    @Override
    public Stream<FriendlyMatchDate> all() {
        return dates.values().stream().sorted(Comparator.comparing(FriendlyMatchDate::date));
    }

    @Override
    public void add(FriendlyMatchDate date) {
        dates.put(date.date(), date);
    }

    @Override
    public Optional<FriendlyMatchDate> byDate(int year, Month month, int dayOfMonth) {
        return Optional.ofNullable(dates.get(LocalDate.of(year, month, dayOfMonth)));
    }
}

File: src/test/kotlin/io/github/binout/soccer/infrastructure/persistence/InMemoryFriendlyMatchDateRepository.kt
package io.github.binout.soccer.infrastructure.persistence

import io.github.binout.soccer.domain.date.FriendlyMatchDate
import io.github.binout.soccer.domain.date.FriendlyMatchDateRepository
import java.time.LocalDate
import java.time.Month
import java.util.concurrent.ConcurrentHashMap

class InMemoryFriendlyMatchDateRepository : FriendlyMatchDateRepository {

    private val dates: MutableMap<LocalDate, FriendlyMatchDate> = ConcurrentHashMap()

    override fun all(): List<FriendlyMatchDate> = dates.values.sortedBy{ it.date }

    override fun add(date: FriendlyMatchDate) {
        dates[date.date] = date
    }

    override fun byDate(year: Int, month: Month, dayOfMonth: Int): FriendlyMatchDate? =
            dates[LocalDate.of(year, month, dayOfMonth)]
}
